deploy/lambda/auto-deployment.js

```diff
@@ -0,0 +1,115 @@
+const https = require('https');
+
+exports.handler = async function (event, context) {
+    if (!process.env.AUTO_DEPLOY_ENABLED) {
+        console.log('Auto Deploying disabled');
+        return;
+    }
+
+    try {
+        const autoDeployUsername = process.env.AUTO_DEPLOY_USERNAME;
+        const autoDeployToken = process.env.AUTO_DEPLOY_TOKEN;
+
+        if (!autoDeployUsername) {
+            console.error('No auto deploy username was provided!');
+        }
+        if (!autoDeployToken) {
+            console.error('No auto deploy token was provided!');
+        }
+
+        const versionURL = `${process.env.PROD_URL}/version`;
+        const mainBranchURL = 'https://api.github.com/repos/atlassian/github-for-jira/branches/main';
+
+        const deployedCommitSHA = (await getRequest(versionURL)).commit;
+        const mainCommitSHA = (await getRequest(mainBranchURL)).commit.sha;
+
+        if (deployedCommitSHA !== mainCommitSHA) {
```

Comment:
/nit - personal preference for me, I would change the `if` to `deployedCommitSHA === mainCommitSHA`then return early and drop the else.

Reply:
Doesn't really make much difference, so leaving it as it is...